### ISSUE TITLE
4 or more equal signs after an ampersand returns an error.

### ISSUE THREAD

**Issue by sabinlinden** [tonybaloney/sphinxcontrib-confluencebuilder#63]
One of my files was generating this error:

```
sphinxcontrib.confluencebuilder error:
---
An unsupported Confluence API call has been made.

REQ: PUT
RSP: 400
URL: https://lindenlab.atlassian.net/wiki/rest/api/
API: content
MSG: Error parsing xhtml: Unexpected character ''' (code 39); expected a semi-colon after the reference for entity 'References'
 at [row,col {unknown-source}]: [49,46]
---
```

This is being triggered by the following text:
```
Research & References
=====================
```

After messing with it for a while, I discovered that the error persists until I delete enough equal signs that the string looks like this:

```
Research & References
===
```

It's specifically at 4 or more equal signs that the error comes back.

**However** if you remove the ampersand without touching the equal signs, like so:

```
Research References
=====================
```

The error goes away. So this has to do with some combination of equal signs and ampersand.

**Comment by jdknight** [comment]
@sabinlinden, by chance, do you have a bigger sample block of data which results in this error?

Using the most recent release, I've tried using the following block of data in a document:
```
Research & References
=====================
```
With the above, I was able to successfully publish to both Confluence Server (6.0.6) and Confluence Cloud. Is there additional data that might be causing the issue?

Other notes:
- Is `Research & References` the first line of the document? If not, is there an empty line above it? Also, is there also an empty line after the equals bar?
- If you're using Confluence Server, might I ask what version you're using?
- Another option you can try is enabling the option `confluence_adv_trace_data` to `True` and observe the generated `trace.log` file. The file should contain a raw output of your publish attempt that you could either share here or manually extract and share the offending line.

**Comment by sabinlinden** [comment]
1: It is not the first line, there is an empty line both above it and there is an empty line below the equals bar.
2: We are using Confluence Cloud.
3: I added
```
confluence_adv_trace_data = True
```
To the conf.py file but no trace.log file was generated. Should it be in the local dir or is it output elsewhere?

**Comment by jdknight** [comment]
@sabinlinden, the file `trace.log` should be generated in the working directory of your session building the documentation. While tracing has been rarely used, I'm a bit perplexed on why you are not seeing any generated log files. :(

If you don't mind also performing a sanity check to ensure you have v0.8 installed:
```
$ python -m sphinxcontrib.confluencebuilder --version
sphinxcontrib.confluencebuilder 0.8
```
Also, mind providing the version of Sphinx you're using?
```
$ sphinx-build --version
(or)
$ python -m sphinx --version
```

**Comment by sabinlinden** [comment]
Haha well first:
```
$ python -m spdsfahinxcontrib.confluencebuilder --version
/usr/bin/python: cannot import name ConfluenceBuilder; 'sphinxcontrib.confluencebuilder' is a package and cannot be directly executed
```

Uh oh to that...
Second:
```
sphinx-build --version
Sphinx (sphinx-build) 1.6.5
```

**Comment by sabinlinden** [comment]
```
$ pip show sphinxcontrib-confluencebuilder
---
Name: sphinxcontrib-confluencebuilder
Version: 0.7.0.dev0
```
Okay, I'm out of date. Let me update and check this then!

**Comment by jdknight** [comment]
Interesting. Out of curiosity, were you dabbling (modifying) anything in the pip's site-package for this extension? I recently tried doing an upgrade of the extension on another environment and noticed that I couldn't query the extension as well (and error such as `ImportError: cannot import name ConfluenceBuilder`); however, I found out that is because I was manually tweaking some stuff for some unrelated testing. If so, it could explain some irregularities. But if that's not the case, I'm wondering if `pip` is not cleaning up some legacy `pyc` files during the upgrade process. I'd be curious if there's another issue masked in the upgrade process (either from pip or something overlooked in this extensions configuration/setup).
